libcoap/src/protocol.rs

```diff
@@ -115,14 +103,20 @@ pub enum CoapOptionType {
     Block1 = COAP_OPTION_BLOCK1 as u16,
     /// Block2 option ([RFC 7959, Section 2.1](https://datatracker.ietf.org/doc/html/rfc7959#section-2.1)).
     Block2 = COAP_OPTION_BLOCK2 as u16,
-    /// Hop-Limit option ([RFC 8768, Section 3](https://datatracker.ietf.org/doc/html/rfc8768#section-3)).
-    HopLimit = COAP_OPTION_HOP_LIMIT as u16,
     /// No-Response option ([RFC 7967, Section 2](https://datatracker.ietf.org/doc/html/rfc7967#section-2)).
     NoResponse = COAP_OPTION_NORESPONSE as u16,
-    /// Observe option ([RFC 7641, Section 2](https://datatracker.ietf.org/doc/html/rfc7641#section-2)).
-    Observe = COAP_OPTION_OBSERVE as u16,
-    // TODO
-    //OsCore = COAP_OPTION_OSCORE as u16,
+    /// OSCORE option ([RFC 8613, Section 2](https://datatracker.ietf.org/doc/html/rfc8613#section-2).
+    OsCore = COAP_OPTION_OSCORE as u16,
```

Comment:
Hmm, it's a matter of taste, I suppose, but I would probably have called this `Oscore`.

Reply:
Yeah, you're right.

Not only a matter of taste, also something the [Rust API Guidelines recommend](https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case).